### PR TITLE
New Template - Create PIP with DDoS Protection

### PR DESCRIPTION
**Pull request by mchakka**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

*
*
*

### CODE REVIEW

**Comment by azure-quickstarts** [comment]
@mchakka - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by azure-quickstarts** [comment]
@mchakka - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by azure-quickstarts** [comment]
@mchakka - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by azure-quickstarts** [comment]
@mchakka - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by mchakka** [comment]
@bmoore-msft do you mind taking a look at this PR and help merging this if it looks good?

File: quickstarts/microsoft.network/pip-with-ddos-ip-protection/main.bicep [hunk 1]
```diff
@@ -0,0 +1,44 @@
+param name string
```
Comment (bmoore-msft) [inline quickstarts/microsoft.network/pip-with-ddos-ip-protection/main.bicep:1]: add description decorators to all params

File: quickstarts/microsoft.network/pip-with-ddos-ip-protection/main.bicep [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  'VirtualNetworkInherited'
+  'Enabled'
+  'Disabled'
+])
+param ddosProtectionMode string = 'Enabled'
+
+@description('Specify a location for the resources.')
+param location string = resourceGroup().location
+
+resource name_resource 'Microsoft.Network/publicIPAddresses@2022-05-01' = {
```
Comment (bmoore-msft) [inline quickstarts/microsoft.network/pip-with-ddos-ip-protection/main.bicep:31]: use a descriptive symbolic name, e.g. `publicIp`

File: quickstarts/microsoft.network/pip-with-ddos-ip-protection/main.bicep [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  name: name
```
Comment (bmoore-msft) [inline quickstarts/microsoft.network/pip-with-ddos-ip-protection/main.bicep:32]: use a descriptive param name, e.g. `publicIpName`

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

**Comment by azure-quickstarts** [comment]
@mchakka - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Comment by azure-quickstarts** [comment]
@mchakka - check this PR for updates that may be needed to documentation that references this sample.  [This is an automated message. You are receiving it because you are listed as the docOwner in metadata.json.]

**Pull request merged**